assets/js/components/read-more/index.js

```diff
@@ -42,52 +65,49 @@ class ReadMore extends Component {
 	 * value of the lines prop. Content is updated once limited.
 	 */
 	clampLines() {
-		const { ellipsis, maxLines } = this.props;
-		const maxHeight = ( this.lineHeight * maxLines ) + 1;
-
-		this.start = 0;
-		this.middle = 0;
-		this.end = this.originalContent.length;
+		const { ellipsis } = this.props;
+		const { maxHeight } = this.state;
+		const originalContent = this.reviewContent.current.innerHTML;
+
+		let markers = {
+			start: 0,
+			middle: 0,
+			end: originalContent.length,
+		};
 
-		while ( this.start <= this.end ) {
-			this.middle = Math.floor( ( this.start + this.end ) / 2 );
-			this.element.current.innerHTML = this.originalContent.slice( 0, this.middle );
+		while ( markers.start <= markers.end ) {
+			markers.middle = Math.floor( ( markers.start + markers.end ) / 2 );
+			this.reviewSummary.current.innerHTML = originalContent.slice( 0, markers.middle );
 
-			if ( this.middle === this.originalContent.length ) {
+			if ( markers.middle === originalContent.length ) {
```

Comment:
We are calling `clampLines()` only when `this.state.isClamped` is set to true, and that only happens when `reviewHeight > maxHeight` so this check `markers.middle === originalContent.length` will always resolve to false and can be removed. Is that right?

Reply:
I think so, will test it